### PR TITLE
feat: [P4PU-226] date format

### PR DESCRIPTION
**Pull request by mmoio**
#### List of Changes
- adds formatDate function, wrapping
  toLocaleString for date formatting
- unit tests

#### How Has This Been Tested?
- visual testing
- unit tests

#### Screenshots (if appropriate):
image

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by mmoio** [comment]
@stratoivandiluccio @vito80ba @AnisLucidi01 I'm not really sure if the predefined Intl is necessary, do we expect to have multiple repeating date? If not i would remove the optimization code for simplicity sake.

File: src/utils/datetools.ts [hunk 1]
```diff
@@ -1,3 +1,37 @@
...
+  if (isNaN(date.getTime())) {
+    return '';
+  }
+
+  // Use Intl instance for performance optimization
+  // on the expected most used locale
+  if (locale === EXPECTED_LOCALE) {
+    return romeDateIntlIt.format(date);
+  } else {
+    return date.toLocaleString(locale, options.romeDate);
```
Comment (stratoivandiluccio) [inline src/utils/datetools.ts:31]: Could you add a test with a locale different from the EXPECTED LOCALE to have this branch covered?

File: src/utils/datetools.ts [hunk 2]
```diff
@@ -1,3 +1,37 @@
...
+const isoToHumanDateRome = (isoDateString?: string, locale = navigator.language) => {
+  if (!isoDateString) {
+    return '';
+  }
+
+  const date = new Date(isoDateString);
+
+  // Check if the date is invalid
+  if (isNaN(date.getTime())) {
+    return '';
```
Comment (stratoivandiluccio) [inline src/utils/datetools.ts:23]: It might be useful, for debugging purposes, to place a console warning here

File: src/utils/datetools.ts (hunk 1 above)
Comment (mmoio) [reply src/utils/datetools.ts:31]: I'm going to remove the optimization code

**Comment by mmoio** [comment]
Force pushed to drop the performance optimization commit and amended adding a console.warning on NaN date.

**Comment by mmoio** [comment]
> @stratoivandiluccio @vito80ba @AnisLucidi01 I'm not really sure if the predefined Intl is necessary, do we expect to have multiple repeating date? If not i would remove the optimization code for simplicity sake.

Removed since we do not expect many duplicated dates.

**Review by AnisLucidi01 — APPROVED** [review]

**Review by stratoivandiluccio — APPROVED** [review]
Looks good to me!

**Pull request merged**